docs/testsuite_overview.md

```diff
@@ -0,0 +1,255 @@
+* wireProtocolTestData - list of Rest.li interface definition files (.restspec.json) as well as test HTTP requests and responses,
+  in the form of files, for operations supported by the interface definition.
+
+##### Test data folders
+These folders contain testing data used by the automated tests in the spec. Most of them were generated by Java Rest.li,
+and should be used when following the spec to test your Rest.li implementation. 
+
+The following folders are included:
+* **data**: input JSON data for testing schemas and json serialization
+* **requests**: correct http requests for wire protocol tests (Rest.li protocol 1)
+* **requests-v2**: correct http requests for wire protocol tests (Rest.li protocol 2)
+* **responses**: input http responses for wire protocol tests (Rest.li protocol 1)
+* **responses-v2**: input http responses for wire protocol tests (Rest.li protocol 2)
+* **restspecs**: restspecs generated from Java resources. These should be used by Rest.li implementations
+to make request builders.
+* **snapshots**: snapshots for resources
+
+### Java TestNG Tests
+In src/test/java, you can find the Java test suite, which uses the TestNG testing framework. 
+You can add test suites for new languages in src/test/. The Java suite is broken into two folders:
+* **test**: Java files containing the tests, and utility methods for running tests. 
+TestRestClientAgainstStandardTestSuite and TestRestClientWithManualAssertions contain the tests,
+while StandardTestSuiteBase has utility methods that load files and compare requests.
+
+* **testsuite**: Java files for building and loading requests and responses.
+
+### Sample Rest.li server 
+The restli-testsuite-server directory contains code for a Rest.li Java server. Using the Java resources and keys,
+Java Rest.li will generate language-independent restspecs in the client-testsuite/restspecs folder. Using these restspecs,
+other Rest.li implementations can generate request builders without modifying server code.
+When adding a new test, you may want to update or add a resource to restli-testsuite-server/src/main/java/testsuite.
+See the section on wire protocol tests in [How To Run the Java TestNG Tests and Expand the Test Suite Specification](testsuite_how_to.md).
+
+Rest.li Test Suite Specification Coverage
+------------------
+The test suite spec is intended to cover:
+
+* **JSON data: Serializing and deserializaing**  
+  Tests cover basic JSON and JSON corner cases such as large numbers, special characters and encodings.
+  
+* **Data Schemas: Generating data templates from schemas**  
+  Tests cover schema types (records, unions, enums, typerefs, ...), primitive types, optionals and defaults.
+  Backward compatibility rules are also covered.
+  
+* **Wire protocol: Building requests and decoding responses**  
+  Tests cover serializing/deserializing of URLs, 
+  Headers and bodies, Escaping/encoding, batch formats, projections, and partial updates.
+
+### JSON Tests
```

Comment:
Prefer to group these section under each bullet above, that will be clearer. For this section, just keep 3 subsections under it. Right now, mixed use of bullets and sub-sections are a bit confusing.